.dependency-cruiser.js

```diff
@@ -183,7 +238,140 @@ module.exports = {
           'npm-peer'
         ]
       }
-    }
+    },
+
+    /* application-specific architecture rules */
+    {
+      name: 'dependencies-of-shared',
+      severity: 'error',
+      from: {
+        path: dir('src/shared')
+      },
+      to: {
+        pathNot: [
+          dir('src/shared'),
+          dir('src/fs'),
+          ...npmDeps(
+            'node-fetch',
```

Comment:
This is an interesting case. As far as I can see, we only import a type from it which is shared by node-fetch and client-side fetch. Maybe we should try to find another solution because node-fetch should actually be a forbidden dependency.

Reply:
No, `src/shared` does not only import types from `node-fetch`: `src/shared/utils/fetch.ts` also imports the `fetch` function. I've added a TODO comment for this.

Reply:
Ah, you are right. I had missed `fetch.ts`. This should really be injected somehow, the current case split is quite dirty:
```
const doFetch = typeof window !== 'undefined' ? window.fetch : fetch;
```